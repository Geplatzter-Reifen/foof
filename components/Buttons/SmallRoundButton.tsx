import React from "react";
import { Button, ButtonProps } from "@ui-kitten/components";
import { RenderProp } from "@ui-kitten/components/devsupport";
import { ImageProps, StyleSheet } from "react-native";

type BigButtonType = Omit<
  React.PropsWithChildren<ButtonProps>,
  "accessoryLeft"
> & { icon: RenderProp<Partial<ImageProps>> | undefined };
export default function SmallRoundButton(props: BigButtonType) {
  return (
    <Button
      {...props}
      style={styles.button}
      accessoryLeft={props.icon}
      onPress={props.onPress}
    ></Button>
  );
}

const styles = StyleSheet.create({
  button: {
    width: 44,
    height: 44,
    borderRadius: 25,
    elevation: 3,
  },
});
