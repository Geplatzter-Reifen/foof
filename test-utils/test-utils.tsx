// https://testing-library.com/docs/react-testing-library/setup/#add-custom-queries
import React from "react";
import { render, RenderOptions } from "@testing-library/react-native";
import { FontAwesomeIconsPack } from "@/components/Font/fontAwesome";
import { mapping } from "@eva-design/eva";
import {
  IconRegistry,
  ApplicationProvider,
  ModalService,
} from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { foofDarkTheme, foofLightTheme } from "@/constants/custom-theme";
import { library } from "@fortawesome/fontawesome-svg-core";
import { far } from "@fortawesome/free-regular-svg-icons";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { fab } from "@fortawesome/free-brands-svg-icons";

// Iconpacks laden
library.add(far, fas, fab);

// ModalService wird verwendet, um die Modal-Positionierung zu steuern
ModalService.setShouldUseTopInsets = true;

type Props = {
  children: React.ReactNode;
  use_dark_theme?: boolean;
};
// Der Wrapper wird verwendet, um die Provider zu umschließen, die für die Komponenten notwendig sind
const AllTheProviders = ({ children, use_dark_theme = true }: Props) => {
  const theme = use_dark_theme
    ? { ...eva.dark, ...foofDarkTheme }
    : { ...eva.light, ...foofLightTheme };
  return (
    <>
      <IconRegistry icons={FontAwesomeIconsPack} />
      <ApplicationProvider {...eva} customMapping={mapping} theme={theme}>
        {children}
      </ApplicationProvider>
    </>
  );
};

// Erstellt einen Custom-Render, der die Provider um die Komponenten wickelt
const customRender = (
  component: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => {
  return render(component, { wrapper: AllTheProviders, ...options });
};

// Exportiert alle Funktionen der Testing Library, weswegen man diese dann auch von dieser Datei importieren kann
export * from "@testing-library/react-native";
export { customRender as render };
