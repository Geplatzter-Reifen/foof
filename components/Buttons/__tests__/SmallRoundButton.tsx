import { render } from "@/test-utils/test-utils";
import SmallRoundButton from "../SmallRoundButton";

describe("SmallRoundButton", () => {
  it("renders correctly", () => {
    const view = render(<SmallRoundButton icon={() => <></>} />);
    expect(view).toMatchSnapshot();
  });
});
