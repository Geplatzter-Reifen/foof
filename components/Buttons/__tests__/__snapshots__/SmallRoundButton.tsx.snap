// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallRoundButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FF6E14",
      "borderColor": "#FF6E14",
      "borderRadius": 25,
      "borderWidth": 1,
      "elevation": 3,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "minHeight": 40,
      "minWidth": 40,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 12,
      "width": 44,
    }
  }
/>
`;
